python/lsst/pipe/supertask/config.py

```diff
@@ -47,6 +49,66 @@
 # ------------------------
 
 
+def makeDatasetField(name, dtype, removeUnitsVariable=False):
+    def factory(**kwargs):
+        return pexConfig.ConfigField(doc=kwargs['doc'],
```

Comment:
The code in this function is being clever so I think it would be helpful to have some comment lines scattered around.